### PR TITLE
Added back methods mistakenly considered unused in QuarkusPluginExtension

### LINKED ISSUE #6311: A generated Kotlin project with Gradle doesn't build
**Describe the bug**
When generating a project like so:

```
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create     -DprojectGroupId=org.acme     -DprojectArtifactId=gradle-kotlin-test     -DclassName="org.acme.GreetingResource"     -Dpath="/greeting"     -Dextensions="kotlin" -DplatformArtifactId=quarkus-bom -DbuildTool=gradle
```

and then trying to build it like so:

```
gradle build -x test
```

I get:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/gandrian/Desktop/gradle-kotlin-test/build.gradle' line: 27

* What went wrong:
A problem occurred evaluating root project 'gradle-kotlin-test'.
> Could not find method setOutputDirectory() for arguments [/home/gandrian/Desktop/gradle-kotlin-test/build/classes/kotlin/main] on object of type io.quarkus.gradle.QuarkusPluginExtension.

```

**Environment (please complete the following information):**
 - Quarkus version or git rev: latest `master`

### PR DESCRIPTION
**Pull request by gastaldi**
These methods were removed in 3265edd082955d6f5b0498dfeb79033c2c073219 because they were considered unused but they are used in the Kotlin Gradle build.
Added tests for all SourceTypes now

Fixes #6311

### CODE REVIEW

**Comment by geoand** [comment]
I tested it and it fixes the original issue, thanks a lot!

**Review by geoand — APPROVED** [review]

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java [hunk 1]
```diff
@@ -60,12 +72,13 @@ public void canBuild(@TempDir File projectRoot) throws IOException {
         return arguments;
     }
 
-    private void createProject(@TempDir File projectRoot) throws IOException {
+    private void createProject(SourceType sourceType) throws IOException {
```
Comment (geoand) [inline devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java]: :+1:

**Comment by gastaldi** [comment]
Looks like we found a bug in the Scala project with Windows (Thanks to the new test!) :)

https://scans.gradle.com/s/turf2ey7xcl2o/tests/dqwcqqtprpdqc-tvfi6t4p2x5we

**Comment by gsmet** [comment]
@gastaldi it looks like an issue similar to the one @jaikiran fixed a few times already: https://github.com/quarkusio/quarkus/pull/6290

**Comment by gastaldi** [comment]
I'll disable the Scala gradle build test for now and open another issue to fix this

**Pull request merged**